From aae860fe3e9f25526a40734a01fa6eadefbe26e6 Mon Sep 17 00:00:00 2001
From: Philip Zander <philip.zander@gmail.com>
Date: Tue, 28 Jul 2020 02:31:52 +0200
Subject: [PATCH] Change  `__declspec(dllexport)` to `extern
 __declspec(dllexport)`

As of 28 July 2020, building on MSYS2 fails with "multiple definition"
errors unless exported variables are explicitly marked as "extern".

Closes #2958
---
 gdk/gdktypes.h     | 2 +-
 gtk/gtkdebug.h     | 2 +-
 gtk/gtkmain.h      | 2 +-
 gtk/gtktexttypes.h | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/gdk/gdktypes.h b/gdk/gdktypes.h
index eed31243ba..c72b20b92e 100644
--- a/gdk/gdktypes.h
+++ b/gdk/gdktypes.h
@@ -39,7 +39,7 @@
 
 #ifdef G_OS_WIN32
 #  ifdef GDK_COMPILATION
-#    define GDKVAR __declspec(dllexport)
+#    define GDKVAR extern __declspec(dllexport)
 #  else
 #    define GDKVAR extern __declspec(dllimport)
 #  endif
diff --git a/gtk/gtkdebug.h b/gtk/gtkdebug.h
index 05964f8dbe..623750301d 100644
--- a/gtk/gtkdebug.h
+++ b/gtk/gtkdebug.h
@@ -64,7 +64,7 @@ typedef enum {
 
 #ifdef G_OS_WIN32
 #  ifdef GTK_COMPILATION
-#    define GTKVAR __declspec(dllexport)
+#    define GTKVAR extern __declspec(dllexport)
 #  else
 #    define GTKVAR extern __declspec(dllimport)
 #  endif
diff --git a/gtk/gtkmain.h b/gtk/gtkmain.h
index 02a1daea9e..2f057e5478 100644
--- a/gtk/gtkmain.h
+++ b/gtk/gtkmain.h
@@ -67,7 +67,7 @@ typedef gint	(*GtkKeySnoopFunc)	    (GtkWidget	  *grab_widget,
  */
 #ifdef G_PLATFORM_WIN32
 #ifdef GTK_COMPILATION
-#define GTKMAIN_C_VAR __declspec(dllexport)
+#define GTKMAIN_C_VAR extern __declspec(dllexport)
 #else
 #define GTKMAIN_C_VAR extern __declspec(dllimport)
 #endif
diff --git a/gtk/gtktexttypes.h b/gtk/gtktexttypes.h
index c624aeaf3a..eb0db36e6d 100644
--- a/gtk/gtktexttypes.h
+++ b/gtk/gtktexttypes.h
@@ -44,7 +44,7 @@ typedef struct _GtkTextMarkBody GtkTextMarkBody;
 
 #ifdef G_OS_WIN32
 #ifdef GTK_COMPILATION
-#define VARIABLE __declspec(dllexport)
+#define VARIABLE extern __declspec(dllexport)
 #else
 #define VARIABLE extern __declspec(dllimport)
 #endif
-- 
GitLab

